File: web/src/core/feature/exchange/domain/entity/CupExchange.ts
export type SupportedExchangeCurrency = "USD" | "EUR";
export type DisplayCurrency = "CUP" | "USD";

export interface CupExchange {
    id: string;
    usdReference: number;
    euroReference: number;
    updatedAt: string;
    source: "DIRECTORIO_CUBANO";
}

export function createCupExchange(cupExchange: CupExchange): CupExchange {
    if(!cupExchange.id || cupExchange.id.trim() === "")
        throw new Error("The value of exchange identifier cannot be empty")
    if (cupExchange.usdReference <= 0 || cupExchange.euroReference <= 0)
        throw new Error("The value of exchange must by positive")
    return {
        ...cupExchange
    }
}

File: app/src/main/java/com/elitec/alejotaller/feature/exchange/domain/entity/CupExchange.kt
package com.elitec.alejotaller.feature.exchange.domain.entity

data class CupExchange(
    val id: String,
    val usdReference: Float,
    val euroReference: Float,
    val updatedAt: String,
    val source: String = "DIRECTORIO_CUBANO"
) {
    init {
        require(usdReference > 0) { "The value of USD exchange must be positive" }
        require(euroReference > 0) { "The value of EUR exchange must be positive" }
        require(id.isNotBlank()) { "The value of exchange identifier cannot be empty" }
    }
}
